packages/supi/test/install/peerDependencies.ts

```diff
@@ -250,6 +251,47 @@ test('top peer dependency is linked on subsequent install', async () => {
   expect(await exists(path.resolve('node_modules/.pnpm/abc-parent-with-ab@1.0.0_peer-c@1.0.0/node_modules/abc-parent-with-ab'))).toBeTruthy()
 })
 
+test('top peer dependency is linked on subsequent install, through transitive peer', async () => {
+  prepareEmpty()
+
+  const manifest = await addDependenciesToPackage({}, ['abc-grand-parent@1.0.0'], await testDefaults())
+
+  await addDependenciesToPackage(manifest, ['peer-c@1.0.0'], await testDefaults())
+
+  expect(await exists(path.resolve('node_modules/.pnpm/abc-grand-parent@1.0.0_peer-c@1.0.0/node_modules/abc-grand-parent'))).toBeTruthy()
+})
+
+test('transitive peers are removed when the subdependency does not rely on them anymore', async () => {
+  const project = prepareEmpty()
```

Comment:
`when the subdependency does not rely on them anymore` not sure what this exactly means